HW5/main.py

```diff
@@ -0,0 +1,55 @@
+import sqlite3
+from collections import namedtuple
+
+from flask import Flask
+from webargs import validate, fields
+from webargs.flaskparser import use_kwargs
+
+app = Flask(__name__)
+
+
+@app.route("/stats_by_city")
+@use_kwargs(
+    {
+        "genre": fields.Str(
+            required=True,
+            validate=validate.Regexp("[a-zA-Z\s\W]")
+        )
+    },
+    location="query"
+)
+def stats_by_city(genre):
+    query = f"""
+        SELECT BillingCountry, max(count)
+        FROM (
+        SELECT genres.Name, invoices.BillingCountry, COUNT(invoices.BillingCountry) as count
+        FROM genres
+        JOIN tracks ON genres.GenreId = tracks.GenreId
+        JOIN invoice_items ON tracks.TrackId = invoice_items.TrackId
+        JOIN invoices ON invoice_items.InvoiceId = invoices.InvoiceId
+        GROUP BY genres.Name, invoices.BillingCountry
+        )
+        WHERE Name='{genre}'
+        """
+
+    country = namedtuple('country', ['name', 'sales'])
+
+    country.name, country.sales = execute_query(query)[0]
```

Comment:
in a case when genre is not in db you will have error, better to add handling of this case

Reply:
There shouldn't be an error. In a case when genre is not in db it will give me _"None"_ and in line 39 I check _"if country.name is None"_.